finagle-example/src/main/scala/com/twitter/finagle/example/zookeeper/ServiceConsumer.scala

```diff
@@ -0,0 +1,42 @@
+package com.twitter.finagle.example.zookeeper
+
+import java.io.ByteArrayInputStream
+import java.nio.charset.StandardCharsets
+
+import com.twitter.finagle.Service
+import com.twitter.finagle.builder.ClientBuilder
+import com.twitter.finagle.example.zookeeper.ServiceProvider.EchoServer
+import com.twitter.finagle.http._
+import com.twitter.io.Reader
+import com.twitter.util.Future
+
+/**
+  * This demo shows how to consume the service in zookeeper.
+  *
+  * Run ServiceProvider before ServiceConsumer
+  *
+  */
+object ServiceConsumer {
+
+  def main(args: Array[String]): Unit = {
+
+    //use zookeeper to discover service
+    val client: Service[Request, Response] = ClientBuilder()
```

Comment:
would you mind using a finagle 6-style client?  This instead becomes:

```scala
Http
  .client
  .withSessionPool.maxSize(10)
  .newService(ServiceProvider.buildConsumerPath(EchoServer.servicePath), "http-client")
```

The new version of finagle that supports the with-style parameters should be released later this week.